pkg/apis/serving/v1beta1/component.go

```diff
@@ -106,6 +107,10 @@ type ComponentExtensionSpec struct {
 	// More info: http://kubernetes.io/docs/user-guide/annotations
 	// +optional
 	Annotations map[string]string `json:"annotations,omitempty"`
+
+	// The deployment strategy to use to replace existing pods with new ones.
+	// +optional
+	DeploymentStrategy *appsv1.DeploymentStrategy `json:"deploymentStrategy,omitempty"`
```

Comment:
What happens when this field is set for the Knative mode?

Reply:
This is only used in `NewDeploymentReconciler.createRawDeployment`

Reply:
the field is still exposed on the isvc which user can set for the knative mode, but I guess knative probably validate this.

Reply:
I think we need to add an validation that this field is not supported for knative otherwise user will expect this to work but knative performs blue/green deployment.

Reply:
Added